### ISSUE TITLE
Py2 vs. Py3 pickling

### ISSUE THREAD

**Issue by tavinathanson** [hammerlab/pyensembl#103]
In https://github.com/hammerlab/pyensembl/pull/99#issuecomment-123836329, @iskander wrote:

>Running the unit tests with Python 2.7 I get different errors:

>```
ERROR: test_all_gene_names : Make sure some known gene names such as
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/Cellar/python/2.7.6/Frameworks/Python.framework/Versions/2.7/lib/python2.7/site-packages/nose-1.3.6-py2.7.egg/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/Users/iskander/code/pyensembl/test/test_common.py", line 35, in new_test_fn
    test_fn(ensembl)
  File "/Users/iskander/code/pyensembl/test/test_gene_names.py", line 23, in test_all_gene_names
    gene_names = ensembl.gene_names()
  File "/Users/iskander/code/pyensembl/pyensembl/common.py", line 56, in wrapped_fn
    value = fn(*args, **kwargs)
  File "/Users/iskander/code/pyensembl/pyensembl/genome.py", line 519, in gene_names
    strand=strand)
  File "/Users/iskander/code/pyensembl/pyensembl/genome.py", line 218, in all_feature_values
    return cached_object(pickle_path, compute_fn=run_query)
  File "/Users/iskander/code/pyensembl/pyensembl/compute_cache.py", line 118, in cached_object
    obj = pickle.load(f)
ValueError: unsupported pickle protocol: 4
```

He addressed this with https://github.com/hammerlab/pyensembl/commit/f3351745a36e54ef23d578c4403c1531da9994ab, though it's still unclear what the underlying issue is. I haven't been able to reproduce locally.

Labeling as a bug for now, but this could be changed to `wontfix` if we can't reproduce it.

**Comment by tavinathanson** [comment]
The issue was simply that we were using `HIGHEST_PROTOCOL` for pickling, and Python 2 doesn't support that protocol.

**Comment by tavinathanson** [comment]
Fixed by #108
